lib/galaxy/webapps/galaxy/api/library_contents.py

```diff
@@ -62,6 +62,24 @@ class JsonApiRoute(APIContentTypeRoute):
 LibraryContentsCreateForm = as_form(LibraryContentsFileCreatePayload)
 
 
+async def get_files(
+    request: Request, trans: ProvidesUserContext = DependsOnTrans, files: Optional[List[UploadFile]] = None
+):
+    # FastAPI's UploadFile is a very light wrapper around starlette's UploadFile
+    if not files:
+        data = await request.form()
+        upload_files = []
+        for upload_file in data.values():
+            if isinstance(upload_file, StarletteUploadFile):
+                with tempfile.NamedTemporaryFile(
+                    dir=trans.app.config.new_file_path, prefix="upload_file_data_", delete=False
+                ) as dest:
+                    shutil.copyfileobj(upload_file.file, dest)  # type: ignore[misc]  # https://github.com/python/mypy/issues/15031
```

Comment:
This is where you're doing I/O

Reply:
So should I move this part into the service file?